### PR TITLE
Feat arrow chart

### PR DESCRIPTION
**Pull request by benib**
- this adds arrow chart as a chart type
- one can configure divering color schemes in process.env.DIVERGING_COLOR_SCHEMES to be selectable via an option
- actual color schemes have to be sent in toolRuntimeConfig then with these configured color scheme names
- vega is updated to version 4 for `vega.schemeDiscretized` function to register discrete color schemes (used to register the diverging scheme before using it)

deployed on test

### CODE REVIEW

File: chartTypes/arrow/mapping.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+                  data.yValue !== null &&
+                  data.yValue !== false &&
+                  data.yValue !== true
+                );
+              })
+              .map((data, index, row) => {
+                // if this is the first data series, we calculate the diff to the next one
+                // this is needed for the annotations.diff options
+
+                // we round the diff to the maximum precision availble in the source data
```
Comment (manuelroth) [inline chartTypes/arrow/mapping.js]: There is a small typo in the comment `availble`

File: chartTypes/arrow/mapping.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+            }
+          }
+        };
+        spec.marks[0].marks[0].marks.push(diffTextMarksSpec);
+      }
+    },
+    {
+      path: "options.annotations.diff",
+      mapToSpec: function(showDiffAnnoation, spec, item) {
+        if (!showDiffAnnoation) {
```
Comment (manuelroth) [inline chartTypes/arrow/mapping.js]: Small typo `showDiffAnnoation` instead of `showDiffAnnotation `

File: routes/rendering-info/web-svg.js [hunk 3]
```diff
@@ -158,6 +163,14 @@ async function getSvg(item, width, toolRuntimeConfig, id, request) {
   return svg;
 }
 
+function registerColorSchemes(type, name, values) {
+  if (type === "discrete") {
+    // this is not working yet because of a bug in vega, the next version should probably fix it
```
Comment (manuelroth) [inline routes/rendering-info/web-svg.js]: Or we now using the version of vega where it works?- If so, this comment can probably be removed

Comment (benib) [reply routes/rendering-info/web-svg.js]: right, removed.

File: resources/fixtures/data/arrow-two-categories-only-negative.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (manuelroth) [inline resources/fixtures/data/arrow-two-categories-only-negative.json]: Maybe it makes sense to also have fixture data for the case with only positive values. What do you think?

Comment (benib) [reply resources/fixtures/data/arrow-two-categories-only-negative.json]: yes, added one in 2f7918854a9747fb492d509598940d29b0239dd8

**Pull request merged**